lib/ansible/modules/network/aci/aci_switch_leaf_policy_profile.py

```diff
@@ -0,0 +1,118 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2017, Bruno Calogero <brunocalogero@hotmail.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+---
+module: aci_switch_leaf_policy_profile
+short_description: Create switch policy leaf profiles on Cisco ACI fabrics (infra:NodeP)
+description:
+- Create switch policy leaf profiles on Cisco ACI fabrics.
+- More information from the internal APIC class
+  I(infra:NodeP) at U(https://developer.cisco.com/site/aci/docs/apis/apic-mim-ref/).
+author:
+- Bruno Calogero (@brunocalogero)
+version_added: '2.5'
+options:
+  switch_profile:
+    description:
+    - The name of the Switch Profile.
+    required: yes
+    aliases: [ switch_profile_name ]
+  description:
+    description:
+    - Description for the Switch Profile.
+  state:
+    description:
+    - Use C(present) or C(absent) for adding or removing.
+    - Use C(query) for listing an object or multiple objects.
+    choices: [ absent, present, query ]
```

Comment:
I wonder if there is a cleaner way of doing `query` @dagwieers any thoughts?

Reply:
We've been doing it like this for all the other modules. What would you be proposing ?

Reply:
Ah, if this is consistent, that I'm happy.